miner/block_builder.go

```diff
@@ -41,7 +41,7 @@ type AtxValidator interface {
 }
 
 type Syncer interface {
-	FetchPoetProof(poetProofRef []byte) error
+	EnsurePoetProof(poetProofRef []byte) error
```

Comment:
Why do you need this refactor? I would like to avoid refactors as much as possible since it makes the cherry-pick to branch `tn` (branch where we test `TN-0.1`) more complicated

Reply:
Because the previous name is misleading IMO. I'll undo it if you don't want refactors.